### PR TITLE
ci(jenkins): enable opbeans syncup

### PR DESCRIPTION
**Pull request by v1v**
When a new tag, aka a new release, is created then, the automation will kick the opbeans build which consists of the following tasks:
- [x] Bump dependency versions in the opbeans-java
- [x] Commit changes. (Implementation within the opbeans-java but the call it's done within this pipeline)
- [x] Push changes.
- [x] Create a tag to trigger the release pipeline in the opbeans-java. That particular tag will match with the same one that the one in the agent itself.

This particular pipeline will push the changes into the `master` branch for the opbeans-go and tag it with the name of the tag version too. Therefore, the opbeans-java pipeline will trigger two builds, the one for its master branch and the other one for the just created tag. This particular flow is implemented within the opbeans-java.

### Checklist

- [x] UTs
- [x] Discuss with the team.

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -262,12 +263,9 @@ pipeline {
     stage('Integration Tests') {
       agent none
       when {
-        beforeAgent true
-        allOf {
-          anyOf {
-            environment name: 'GIT_BUILD_CAUSE', value: 'pr'
-            expression { return !params.Run_As_Master_Branch }
-          }
+        anyOf {
```
Comment (v1v) [inline Jenkinsfile]: Simplify the when condition. This is outside of the scope of this particular PR

File: Jenkinsfile [hunk 2]
```diff
@@ -281,6 +279,31 @@ pipeline {
...
     }
+    stage('Release') {
+      options { skipDefaultCheckout() }
+      when {
+        tag pattern: 'v\\d+\\.\\d+\\d+', comparator: 'REGEXP'
+      }
+      stages {
+        stage('Opbeans') {
+          environment {
+            REPO_NAME = "${OPBEANS_REPO}"
```
Comment (v1v) [inline Jenkinsfile]: This is required to use the git*** steps as it does infer them under the hood

File: Jenkinsfile [hunk 3]
```diff
@@ -281,6 +279,31 @@ pipeline {
...
+          }
+          steps {
+            deleteDir()
+            dir("${OPBEANS_REPO}"){
+              git credentialsId: 'f6c7695a-671e-4f4f-a331-acdce44ff9ba',
+                  url: "git@github.com:elastic/${OPBEANS_REPO}.git"
+              sh script: ".ci/bump-version.sh ${env.BRANCH_NAME}", label: 'Bump version'
```
Comment (v1v) [inline Jenkinsfile]: BRANCH_NAME will match the tag release, as the when condition https://github.com/elastic/apm-agent-java/pull/940/files#diff-58231b16fdee45a03a4ee3cf94a9f2c3R285 filters what tags should trigger this particular stage

File: Jenkinsfile [hunk 4]
```diff
@@ -281,6 +279,31 @@ pipeline {
...
+              // The opbeans-go pipeline will trigger a release for the master branch
```
Comment (v1v) [inline Jenkinsfile]:
```suggestion
              // The opbeans-java pipeline will trigger a release for the master branch
```

File: Jenkinsfile [hunk 5]
```diff
@@ -281,6 +279,31 @@ pipeline {
...
+              gitPush()
+              // The opbeans-go pipeline will trigger a release for the release tag
```
Comment (v1v) [inline Jenkinsfile]:
```suggestion
              // The opbeans-java pipeline will trigger a release for the release tag
```

File: Jenkinsfile [hunk 6]
```diff
@@ -282,6 +280,32 @@ pipeline {
...
+          environment {
+            REPO_NAME = "${OPBEANS_REPO}"
+          }
+          steps {
+            deleteDir()
+            dir("${OPBEANS_REPO}"){
+              git credentialsId: 'f6c7695a-671e-4f4f-a331-acdce44ff9ba',
+                  url: "git@github.com:elastic/${OPBEANS_REPO}.git"
+              // It's required to transform the tag value to the artifact version
+              sh script: ".ci/bump-version.sh ${env.BRANCH_NAME.replaceAll('^v', '')}", label: 'Bump version'
```
Comment (v1v) [inline Jenkinsfile]: There are two approaches as far as I see:
- Grab the `VERSION` from the pom.xml
- Edit the `BRANCH_NAME` and replace `v` at the beginning.

File: Jenkinsfile [hunk 7]
```diff
@@ -282,6 +280,32 @@ pipeline {
         githubNotify(context: "${env.GITHUB_CHECK_ITS_NAME}", description: "${env.GITHUB_CHECK_ITS_NAME} ...", status: 'PENDING', targetUrl: "${env.JENKINS_URL}search/?q=${env.ITS_PIPELINE.replaceAll('/','+')}")
       }
     }
+    stage('Release') {
```
Comment (felixbarny) [inline Jenkinsfile]:
```suggestion
    stage('AfterRelease') {
```

this stage is not actually performing the release
